Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1060,34 +1109,37 @@ bool GCMemcard::ReadBannerRGBA8(u8 index, u32* buffer) const
   if (flags == 0xFB)
     flags = ~flags;
 
-  int bnrFormat = (flags & 3);
+  const int bnr_format = (flags & 3);
 
-  if (bnrFormat == 0)
+  if (bnr_format == 0)
     return false;
 
-  u32 DataOffset = GetActiveDirectory().m_dir_entries[index].m_image_offset;
-  u32 DataBlock = GetActiveDirectory().m_dir_entries[index].m_first_block - MC_FST_BLOCKS;
-
-  if ((DataBlock > m_size_blocks) || (DataOffset == 0xFFFFFFFF))
-  {
+  const u32 offset = GetActiveDirectory().m_dir_entries[index].m_image_offset;
+  if (offset == 0xFFFFFFFF)
     return false;
-  }
 
-  const int pixels = 96 * 32;
+  const u32 pixel_count = 96 * 32;
+  const u32 palette_indices = 256;
+  const bool is_ci8 = (bnr_format & 1) != 0;
```

Comment:
I wonder if this is really 4 possible values (0..3) and not just two bit flags (first one for 8/16 bit, second one for direct/indexed color; with both zero indicating a special "none" value).
YAGCD documents the former, but how we use it here practically means the values 1 and 3 aren't any different (which also ties into the comment down below for the animation).